lib/elixir/lib/keyword.ex

```diff
@@ -946,10 +946,15 @@ defmodule Keyword do
   end
 
   @doc """
-  Returns and removes all values associated with `key` in the keyword list.
+  Returns the value for `key` and removes all associated entries in the keyword list.
 
-  All duplicated keys are removed. See `pop_first/3` for
-  removing only the first entry.
+  It returns a tuple where the first value for `key` is the first element and the
+  second element is a keyword list with all values associated with `key` are removed.
```

Comment:
The combination of "with + are" feels weird to me.

```suggestion
  second element is a keyword list with all entries associated with `key` removed.
```

Or, as an alternative, "a keyword list in which all entried associated with `key` are removed."